framework/db/Command.php

```diff
@@ -419,6 +419,39 @@ public function insert($table, $columns)
     }
 
     /**
+     * Creates an INSERT INTO RETURNING command or plain INSERT if the underlying DBMS does not support it.
+     * @see insert()
+     *
+     * If $returnParams is not empty after calling this, those params should be bound to the query.
+     *
+     * Note that the created command is not executed until [[execute()]] is called.
+     *
+     * @param string $table the table that new rows will be inserted into.
+     * @param array $columns the column data (name => value) to be inserted into the table.
+     * @param array $returnColumns the column names which values to be returned after inserting rows.
+     * @param array $returnParams if not empty after calling this method, each value is bound to the query.
+     * @return Command the command object itself
+     * @throws NotSupportedException if $returnColumns is set but the underlying DBMS doesn't support returning
+     */
+    public function insertReturning($table, $columns, $returnColumns = null, &$returnParams = null)
+    {
+        $params = [];
+        $sql = $this->db->getQueryBuilder()->insertReturning($table, $columns, $params, $returnColumns, $returnParams);
+
+        $this->setSql($sql)->bindValues($params);
+
+        if ($returnParams !== null) {
+            $this->prepare(false);
+            foreach ($returnParams as $param => &$value) {
+                //! @todo type has to be determined from column schema
```

Comment:
Anybody got any idea how to implement this? Actually we could detect only integer/numeric and just use string for everything else, but maybe some valid `column type => pdo type` map could be created.